pyopencl/__init__.py

```diff
@@ -445,6 +451,17 @@ def __getattr__(self, attr):
             # but this will give an error if the kernel is invalid.
             knl.num_args  # noqa: B018
 
+            count = self._knl_retrieval_count[attr] = (
+                self._knl_retrieval_count.get(attr, 0) + 1)
+
+            if count > 1:
+                warn(f"Kernel '{attr}' has been retrieved more than once. "
```

Comment:
These warnings will generally be deduplicated, but in light of https://github.com/inducer/pymbolic/pull/147, would it make sense just to issue the warning once (and maybe use a bool flag, not an int)?